File: in_Java/0476 Number Complement.java
class Solution {
    public int findComplement(int num) {
        int res = 0;
        int p = 0;
        int t;
        while(num != 0){
            t = num % 2;
            if(t == 0){
                res += Math.pow(2,p);
            }
            p++;
            num /= 2;
        }
        return res;
    }
}

File: in_Python/0476 Number Complement.py
class Solution(object):
    def findComplement(self, num):
        """
        :type num: int
        :rtype: int
        """
        res = 0
        idx = 0
        while num != 0:
            c = num % 2
            num /= 2
            res += (1-c) * pow(2, idx)
            idx += 1
        return res